worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -354,67 +381,97 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
     if (healthyWorkingDirs().size <= 0 && !hasHDFSStorage) {
       throw new IOException("No available working dirs!")
     }
-    val shuffleKey = Utils.makeShuffleKey(appId, shuffleId)
-    val (flusher, diskFileInfo, workingDir) = createFile(
+    val partitionDataWriterContext = new PartitionDataWriterContext(
+      splitThreshold,
+      splitMode,
+      rangeReadFilter,
       location,
       appId,
       shuffleId,
-      location.getFileName,
       userIdentifier,
       partitionType,
       partitionSplitEnabled)
+
     val writer =
       try {
         partitionType match {
           case PartitionType.MAP => new MapPartitionDataWriter(
               this,
-              diskFileInfo,
-              flusher,
               workerSource,
               conf,
               deviceMonitor,
-              splitThreshold,
-              splitMode,
-              rangeReadFilter,
-              shuffleKey)
+              partitionDataWriterContext)
           case PartitionType.REDUCE => new ReducePartitionDataWriter(
               this,
-              diskFileInfo,
-              flusher,
               workerSource,
               conf,
               deviceMonitor,
-              splitThreshold,
-              splitMode,
-              rangeReadFilter,
-              shuffleKey)
+              partitionDataWriterContext)
           case _ => throw new UnsupportedOperationException(s"Not support $partitionType yet")
         }
       } catch {
         case e: Exception =>
           logError("Create partition data writer failed", e)
           throw e
       }
-    if (!(writer.getDiskFileInfo.isHdfs)) {
-      deviceMonitor.registerFileWriter(writer)
-      workingDirWriters.computeIfAbsent(workingDir, workingDirWriterListFunc).put(
-        diskFileInfo.getFilePath,
-        writer)
-    } else {
-      hdfsWriters.put(diskFileInfo.getFilePath, writer)
-    }
     writer
   }
 
-  def getDiskFileInfo(shuffleKey: String, fileName: String): DiskFileInfo = {
-    val shuffleMap = diskFileInfos.get(shuffleKey)
-    if (shuffleMap ne null) {
-      shuffleMap.get(fileName)
-    } else {
-      null
+  def registerMemoryPartitionWriter(writer: PartitionDataWriter, fileInfo: MemoryFileInfo): Unit = {
+    memoryWriters.put(fileInfo, writer)
+  }
+
+  def unregisterMemoryPartitionWriterAndFileInfo(
+      fileInfo: MemoryFileInfo,
+      shuffleKey: String,
+      fileName: String): Unit = {
+    memoryFileInfos.synchronized {
```

Comment:
what if fetchHandler get the MemoryFIleInfo for reader but MemoryManger trigger evict?